src/Microsoft.Azure.WebJobs.Host/Blobs/BlobPath.cs

```diff
@@ -55,16 +55,32 @@ public static BlobPath ParseAndValidate(string value, bool isContainerBinding =
             return path;
         }
 
-        public static BlobPath Parse(string value, bool isContainerBinding)
+        // throws exception if failed
+        public static string ConvertAbsUrlToContainerBlob(string blobUrl)
```

Comment:
>ConvertAbsUrlToContainerBlob [](start = 29, length = 28)

This approach feels wrong. We should create the CloudBlob around the URI directly. That would support sas urls, etc.